packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -964,21 +1083,27 @@ class _RenderSegmentedControl<T extends Object> extends RenderBox
 
   @override
   Size computeDryLayout(BoxConstraints constraints) {
-    final Size childSize = _calculateChildSize(constraints);
-    return _computeOverallSizeFromChildSize(childSize, constraints);
+    return _computeOverallSizeFromChildSize(constraints);
   }
 
   @override
   void performLayout() {
     final BoxConstraints constraints = this.constraints;
-    final Size childSize = _calculateChildSize(constraints);
-    final BoxConstraints childConstraints = BoxConstraints.tight(childSize);
-    final BoxConstraints separatorConstraints = childConstraints.heightConstraints();
+    final List<double> childWidths;
+    if (isProportionalSegment) {
+      childWidths = _getChildIntrinsicWidths(constraints);
+    } else {
+      final Size childSize = _calculateChildSize(constraints);
+      childWidths = List<double>.filled(childCount, childSize.width);
+    }
 
+    final double childHeight = _getMaxChildHeight(constraints, double.infinity);
     RenderBox? child = firstChild;
     int index = 0;
     double start = 0;
     while (child != null) {
+      final BoxConstraints childConstraints = BoxConstraints.tight(Size(childWidths[index ~/ 2], childHeight));
```

Comment:
every child should have the same height I think?

Reply:
Moved seperator constraints out of the while loop!

Reply:
oh the separators. Nvm.